packages/xo-server/src/api/pool.mjs

```diff
@@ -1,11 +1,45 @@
 import { asyncMap } from '@xen-orchestra/async-map'
+import { createLogger } from '@xen-orchestra/log'
+import { createPredicate } from 'value-matcher'
 import { defer as deferrable } from 'golike-defer'
 import { format } from 'json-rpc-peer'
 import { Ref } from 'xen-api'
-import { incorrectState } from 'xo-common/api-errors.js'
+import { forbiddenOperation, incorrectState } from 'xo-common/api-errors.js'
 
 import { moveFirst } from '../_moveFirst.mjs'
 
+const log = createLogger('xo:pool')
+
+// ===================================================================
+
+export async function backupGuard(poolId) {
+  const jobs = await this.getAllJobs()
+  const guard = id => {
+    if (this.getObject(id).$poolId === poolId) {
+      throw forbiddenOperation('Backup is running', `A backup is running on the pool: ${poolId}`)
+    }
+  }
+
+  jobs.forEach(({ runId, type, vms }) => {
+    if (runId !== undefined && type === 'backup') {
+      if (vms.id !== undefined) {
+        const id = vms.id.__or ?? vms.id
+        if (Array.isArray(id)) {
+          id.forEach(guard)
+        } else {
+          guard(id)
+        }
+      } else {
+        // smartmode
+        const isPoolSafe = vms.$pool === undefined ? false : !createPredicate(vms.$pool)(poolId)
```

Comment:
I don't understand why pool is unsafe if vms.$pool is `undefined`

Reply:
In case a backup job is created with the `smartmode`, and the user has not defined `Resident on` and `Not resident on` then, the smartmode will consider this as: `Resident on: all pool`